src/drm/drm_handler.js

```diff
@@ -0,0 +1,128 @@
+  + '</PlayReadyInitiator>'
+  return template
+}
+
+export const createDrmAgent = () => {
+  const drmElement = document.createElement('object')
+
+  drmElement.id = 'oipfdrmagent'
+  drmElement.type = 'application/oipfdrmagent'
+  drmElement.style.visibility = 'hidden'
+  drmElement.style.position = 'absolute'
+  drmElement.style.top = '0'
+  drmElement.style.left = '0'
+  drmElement.style.width = '0'
+  drmElement.style.height = '0'
+
+  return drmElement
+}
+
+/* eslint-disable-next-line func-style */
+export function sendLicenseRequest(config = {}, callback = () => {}) {
+  const successCallback = callback.bind(this)
+  let oipfdrmagent = document.getElementById('oipfdrmagent')
+
+  if (!oipfdrmagent) {
+    oipfdrmagent = createDrmAgent()
+    this && this.el
+      ? this.el.appendChild(oipfdrmagent)
+      : document.body.appendChild(oipfdrmagent)
+  }
+
+  !config.licenseServerURL
+    && Log.warn('DRMHandler', 'No one license server is found. The expected result for this behavior is clear the current license.')
+  const xmlLicenceAcquisition = getLicenseOverrideMessageTemplate(config.licenseServerURL)
+
+  const drmRightsErrorHandler = resultCode => {
+    const errorMessage = {
+      0: 'DRM: No license error',
+      1: 'DRM: Invalid license error',
+      2: 'license valid',
+    }
+
+    if (resultCode < 2) {
+      Log.error('DRMHandler', 'Error at onDRMRightsError call', errorMessage[resultCode])
+      return errorMessage[resultCode]
+    }
+  }
+
+  const drmMessageResultHandler = (messageID, resultMessage, resultCode) => {
+    Log.info('DRMHandler', 'onDRMMessageResult messageID', messageID)
+    Log.info('DRMHandler', 'onDRMMessageResult resultMessage', resultMessage)
+    Log.info('DRMHandler', 'onDRMMessageResult resultCode', resultCode)
+
+    const errorMessage = {
+      1: 'DRM: Unspecified error',
+      2: 'DRM: Cannot process request',
+      3: 'DRM: Wrong format',
+      4: 'DRM: User Consent Needed',
+      5: 'DRM: Unknown DRM system',
+    }
+
+    if (resultCode > 0) {
+      Log.error('DRMHandler', 'Error at onDRMMessageResult call', errorMessage[resultCode])
+      return errorMessage[resultCode]
+    }
+
+    successCallback()
+  }
+
+  try {
+    oipfdrmagent.onDRMRightsError = drmRightsErrorHandler
+    oipfdrmagent.onDRMMessageResult = drmMessageResultHandler
+    oipfdrmagent.sendDRMMessage(MESSAGE_TYPE, xmlLicenceAcquisition, DRM_SYSTEM_ID)
+  } catch (error) {
+    Log.error('DRMHandler', 'Error at sendDRMMessage call', error.message)
+  }
+}
+
+/* eslint-disable-next-line func-style */
+export function clearLicenseRequest(callback = () => {}) {
+  const successCallback = callback.bind(this)
+  const oipfdrmagent = document.getElementById('oipfdrmagent')
+
+  if (!oipfdrmagent) {
+    Log.warn('DRMHandler', 'No one DRM license has been configured before. It\'s not necessary to clear any license server.')
+    return
+  }
+
+  const xmlLicenceAcquisition = getClearMessageTemplate()
```

Comment:
Do you have links for reference to the playready message templates?

Reply:
The documents that helped me at this point are WebOS: https://webostv.developer.lge.com/api/webos-service-api/drm/drm-message/#playready